### PR TITLE
Content cli reworkings

### PR DESCRIPTION
**Pull request by iorveth**
Closes https://github.com/Joystream/joystream/issues/2237

### CODE REVIEW

File: cli/src/base/ContentDirectoryCommandBase.ts [hunk 1]
```diff
@@ -85,14 +67,14 @@ export default abstract class ContentDirectoryCommandBase extends RolesCommandBa
...
 
-    return createType('Actor', { Curator: [groupId, curator.workerId.toNumber()] })
+    return createType('ContentActor', { Curator: [groupId, curator.workerId.toNumber()] })
   }
 
-  async promptForClass(message = 'Select a class'): Promise<Class> {
-    const classes = await this.getApi().availableClasses()
-    const choices = classes.map(([, c]) => ({ name: c.name.toString(), value: c }))
+  async promptForChannel(message = 'Select a channel'): Promise<Channel> {
+    const classes = await this.getApi().availableChannels()
```
Comment (Lezek123) [inline cli/src/base/ContentDirectoryCommandBase.ts]: Should be renamed to `channels`

File: cli/src/Api.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export const apiModuleByGroup: { [key in WorkingGroups]: string } = {
 // Api wrapper for handling most common api calls and allowing easy API implementation switch in the future
 export default class Api {
   private _api: ApiPromise
-  private _cdClassesCache: [ClassId, Class][] | null = null
+  private _cdClassesCache: [ChannelId, Channel][] | null = null
```
Comment (Lezek123) [inline cli/src/Api.ts]: Should be renamed to `_cdChannelsCache` (assuming the cache is even still needed)

File: cli/src/Api.ts [hunk 3]
```diff
@@ -495,41 +495,41 @@ export default class Api {
...
-  async entityById(id: number): Promise<Entity | null> {
-    const exists = !!(await this._api.query.contentDirectory.entityById.size(id)).toNumber()
-    return exists ? await this._api.query.contentDirectory.entityById<Entity>(id) : null
+  async videoById(videoId: number): Promise<Video | null> {
+    const exists = !!(await this._api.query.content.entityById.size(videoId)).toNumber()
+    return exists ? await this._api.query.content.videoById<Video>(videoId) : null
   }
 
-  async dataByContentId(contentId: ContentId): Promise<DataObject | null> {
+  async dataByContentId(contentId: number): Promise<DataObject | null> {
```
Comment (Lezek123) [inline cli/src/Api.ts]: `contentId` is represented as either `ContentId` or `string`

File: cli/src/Api.ts [hunk 4]
```diff
@@ -495,41 +495,41 @@ export default class Api {
...
     const dataObject = await this._api.query.dataDirectory.dataByContentId<Option<DataObject>>(contentId)
```
Comment (Lezek123) [inline cli/src/Api.ts]: The query now returns `DataObject` (instead of `Option<DataObject>`) - to check whether it actually exists you'd need to use either `!dataObject.isEmpty` or `!!(await this._api.query.dataDirectory.dataByContentId(contentId).size())`

File: cli/src/base/ContentDirectoryCommandBase.ts [hunk 5]
```diff
@@ -101,11 +83,11 @@ export default abstract class ContentDirectoryCommandBase extends RolesCommandBa
     return selectedClass
   }
 
-  async classEntryByNameOrId(classNameOrId: string): Promise<[ClassId, Class]> {
-    const classes = await this.getApi().availableClasses()
-    const foundClass = classes.find(([id, c]) => id.toString() === classNameOrId || c.name.toString() === classNameOrId)
+  async channelEntryById(channelId: string): Promise<[ChannelId, Channel]> {
```
Comment (Lezek123) [inline cli/src/base/ContentDirectoryCommandBase.ts]: I think now you can just use `channelById` api query, this was just not possible before with classes.

File: yarn.lock [hunk 6]
```diff
@@ -9118,6 +9118,19 @@ cli-cursor@^3.1.0:
   dependencies:
     restore-cursor "^3.1.0"
 
+<<<<<<< HEAD
```
Comment (Lezek123) [inline yarn.lock]: `yarn.lock` was committed with merge conflicts which caused the CI checks to fail instantly.
There are a few prettier/linter issues (most of them can be fixed with `yarn workspace @joystream/cli format`)

**Review by Lezek123 — CHANGES_REQUESTED** [review]

File: content-metadata-protobuf/proto/Channel.proto [hunk 7]
```diff
@@ -1,5 +1,7 @@
 syntax = "proto2";
 
+package muthu.other;
```
Comment (Lezek123) [inline content-metadata-protobuf/proto/Channel.proto]: Also - I'm not sure what is the reason behind this change. Is this the reason why there are so many diffs in compiled protobuffs?

**_EDIT:_** I think think the compiled files were changed probably due to different version of `protoc`, can you share which version you've been using?

Comment (iorveth) [reply content-metadata-protobuf/proto/Channel.proto]: libprotoc 3.6.1

File: yarn.lock (hunk 6 above)
Comment (iorveth) [reply yarn.lock]: done here https://github.com/Joystream/joystream/pull/2258/commits/91fa8e99eb0d25e4f1f2b2bbb6b504860634bfc3

File: cli/src/base/ContentDirectoryCommandBase.ts (hunk 5 above)
Comment (iorveth) [reply cli/src/base/ContentDirectoryCommandBase.ts]: Yeah, removed it in further PR, done here https://github.com/Joystream/joystream/commit/91fa8e99eb0d25e4f1f2b2bbb6b504860634bfc3#diff-039a3501a5eb9ac3c0187a6ed3208e5881dda8ac0d93a52d158913b4b85b05d7L86

File: cli/src/Api.ts (hunk 4 above)
Comment (iorveth) [reply cli/src/Api.ts]: Done, thank you for pointing that issue https://github.com/Joystream/joystream/commit/91fa8e99eb0d25e4f1f2b2bbb6b504860634bfc3#diff-a7a049ad43ba0426653943dc5211d8aea8c49e82a5ecef553d1aa18e7cf48008R529

File: cli/src/Api.ts (hunk 3 above)
Comment (iorveth) [reply cli/src/Api.ts]: Renamed it back to ContentId here https://github.com/Joystream/joystream/commit/91fa8e99eb0d25e4f1f2b2bbb6b504860634bfc3#diff-a7a049ad43ba0426653943dc5211d8aea8c49e82a5ecef553d1aa18e7cf48008R529

File: cli/src/Api.ts (hunk 2 above)
Comment (iorveth) [reply cli/src/Api.ts]: done https://github.com/Joystream/joystream/commit/91fa8e99eb0d25e4f1f2b2bbb6b504860634bfc3#diff-a7a049ad43ba0426653943dc5211d8aea8c49e82a5ecef553d1aa18e7cf48008L70

File: cli/src/base/ContentDirectoryCommandBase.ts (hunk 1 above)
Comment (iorveth) [reply cli/src/base/ContentDirectoryCommandBase.ts]: done https://github.com/Joystream/joystream/commit/91fa8e99eb0d25e4f1f2b2bbb6b504860634bfc3#diff-a7a049ad43ba0426653943dc5211d8aea8c49e82a5ecef553d1aa18e7cf48008L70

File: content-metadata-protobuf/proto/Channel.proto (hunk 7 above)
Comment (Lezek123) [reply content-metadata-protobuf/proto/Channel.proto]: Thanks. I think the buffers were previously compiled using a newer version. I'm currently using `libprotoc 3.15.6` and I'm getting results similiar to those from before this PR. Perhaps those changes can be reverted?

File: yarn.lock (hunk 6 above)
Comment (Lezek123) [reply yarn.lock]: Looks like this one wasn't committed?

File: cli/src/base/ContentDirectoryCommandBase.ts (hunk 1 above)
Comment (Lezek123) [reply cli/src/base/ContentDirectoryCommandBase.ts]: I couldn't find it in the diffs, perhpas it didn't get committed?

File: content-metadata-protobuf/proto/Channel.proto (hunk 7 above)
Comment (iorveth) [reply content-metadata-protobuf/proto/Channel.proto]: Jsut compiled them with libprotoc 3.15.6, the diffs looks the same now, will update PR shortly after resolving `yarn.lock` issues

File: yarn.lock (hunk 6 above)
Comment (iorveth) [reply yarn.lock]: Done https://github.com/Joystream/joystream/pull/2258/commits/2db5a7d3e8fe9d2e50aa90b0febe4be3b8dd03f2

File: cli/src/base/ContentDirectoryCommandBase.ts (hunk 1 above)
Comment (iorveth) [reply cli/src/base/ContentDirectoryCommandBase.ts]: really weird, i can't commit it anymore, though i didn't get any error and can't find it in `.gitignore `

Comment (iorveth) [reply cli/src/base/ContentDirectoryCommandBase.ts]: done

Comment (iorveth) [reply cli/src/base/ContentDirectoryCommandBase.ts]: https://github.com/Joystream/joystream/pull/2258/commits/988403a09592e55af96facff55ba7f30cec1da22

**Review by Lezek123 — APPROVED** [review]
The checks show some small prettier/linter issues, I would suggest using eslint plugin for vscode with autofix on save and/or running the checks (ie. `yarn workspace @joystream/cli checks`) locally to see if there are any errors.

Other than that I think it's good to go!

**Comment by iorveth** [comment]
> The checks show some small prettier/linter issues, I would suggest using eslint plugin for vscode with autofix on save and/or running the checks (ie. `yarn workspace @joystream/cli checks`) locally to see if there are any errors.
> 
> Other than that I think it's good to go!

Done

**Comment by mnaamani** [comment]
merge into https://github.com/Joystream/joystream/pull/2334
